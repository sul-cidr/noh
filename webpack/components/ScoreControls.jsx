import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { setScoreToggles } from "../actionCreators";
import TimelineIndicator from "./TimelineIndicator";

class ScoreControls extends Component {
  constructor(props) {
    super(props);
    const {
      isBeatOn,
      isTextOn,
      isPercussionOn,
      isNohkanOn,
      isDanceOn,
      isPrevSentenceOn,
      isNextSentenceOn
    } = this.props;
    this.state = {
      isBeatOn,
      isTextOn,
      isPercussionOn,
      isNohkanOn,
      isDanceOn,
      isPrevSentenceOn,
      isNextSentenceOn
    };
    this.filtersPopup = null;
    this.handleToggle = this.handleToggle.bind(this);
  }

  componentDidMount() {
    this.props.updateScoreToggles(this.props);
  }

  componentDidUpdate() {
    this.props.updateScoreToggles(this.state);
  }

  handleToggle(event, toggleName) {
    this.setState({
      [toggleName]: event.target.checked
    });
  }

  render() {
    return (
      <div className="score-controls">
        <div className="sentence-control">
          <div className="sentence-control__prev">
            <i className="fas fa-step-backward" />
          </div>
          <div className="sentence-control__status">
            <span className="sentence-control__title">Sentence:</span>
            <span className="sentence-control__current">13/55</span>
          </div>
          <div className="sentence-control__next">
            <i className="fas fa-step-forward" />
          </div>
        </div>
        <div className="video-progress">
          <TimelineIndicator
            startTime={this.props.startTime}
            duration={this.props.duration}
          />
        </div>
        <div className="score-controls__filters">
          <button
            className="score-controls__filters-button"
            onClick={() => this.filtersPopup.classList.toggle("hidden")}
          >
            Filters
          </button>
          <div
            className="score-controls__filters-popup hidden"
            ref={filtersPopup => {
              this.filtersPopup = filtersPopup;
            }}
          >
            <ul className="channel-toggles">
              <li className="toggles__title">Lines</li>
              <li>
                <div className="custom-checkbox">
                  <input
                    id="scoreBeat"
                    type="checkbox"
                    checked={this.state.isBeatOn}
                    onChange={event => this.handleToggle(event, "isBeatOn")}
                    onKeyPress={null}
                  />
                  <label htmlFor="scoreBeat">
                    <span className="custom-checkbox__text">Beat</span>
                  </label>
                </div>
              </li>
              <li>
                <div className="custom-checkbox">
                  <input
                    id="scoreText"
                    type="checkbox"
                    checked={this.state.isTextOn}
                    onChange={event => this.handleToggle(event, "isTextOn")}
                    onKeyPress={null}
                  />
                  <label htmlFor="scoreText">
                    <span className="custom-checkbox__text">Text</span>
                  </label>
                </div>
              </li>
              <li>
                <div className="custom-checkbox">
                  <input
                    id="scorePercussion"
                    type="checkbox"
                    checked={this.state.isPercussionOn}
                    onChange={event =>
                      this.handleToggle(event, "isPercussionOn")
                    }
                    onKeyPress={null}
                  />
                  <label htmlFor="scorePercussion">
                    <span className="custom-checkbox__text">Percussion</span>
                  </label>
                </div>
              </li>
              <li>
                <div className="custom-checkbox">
                  <input
                    id="scoreNohkan"
                    type="checkbox"
                    checked={this.state.isNohkanOn}
                    onChange={event => this.handleToggle(event, "isNohkanOn")}
                    onKeyPress={null}
                  />
                  <label htmlFor="scoreNohkan">
                    <span className="custom-checkbox__text">Nohkan</span>
                  </label>
                </div>
              </li>
              <li>
                <div className="custom-checkbox">
                  <input
                    id="scoreDance"
                    type="checkbox"
                    checked={this.state.isDanceOn}
                    onChange={event => this.handleToggle(event, "isDanceOn")}
                    onKeyPress={null}
                  />
                  <label htmlFor="scoreDance">
                    <span className="custom-checkbox__text">Dance</span>
                  </label>
                </div>
              </li>
            </ul>
            <ul className="measure-toggles">
              <li className="toggles__title">Sentences</li>
              <li>
                <div className="custom-checkbox">
                  <input
                    id="scorePrevSentence"
                    type="checkbox"
                    checked={this.state.isPrevSentenceOn}
                    onChange={event =>
                      this.handleToggle(event, "isPrevSentenceOn")
                    }
                    onKeyPress={null}
                  />
                  <label htmlFor="scorePrevSentence">
                    <span className="custom-checkbox__text">
                      Previous Sentence
                    </span>
                  </label>
                </div>
              </li>
              <li>
                <div className="custom-checkbox">
                  <input
                    id="scoreNextSentence"
                    type="checkbox"
                    checked={this.state.isNextSentenceOn}
                    onChange={event =>
                      this.handleToggle(event, "isNextSentenceOn")
                    }
                    onKeyPress={null}
                  />
                  <label htmlFor="scoreNextSentence">
                    <span className="custom-checkbox__text">Next Sentence</span>
                  </label>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    );
  }
}

ScoreControls.propTypes = {
  isBeatOn: PropTypes.bool,
  isTextOn: PropTypes.bool,
  isPercussionOn: PropTypes.bool,
  isNohkanOn: PropTypes.bool,
  isDanceOn: PropTypes.bool,
  isPrevSentenceOn: PropTypes.bool,
  isNextSentenceOn: PropTypes.bool,
  updateScoreToggles: PropTypes.func,
  startTime: PropTypes.number,
  duration: PropTypes.number.isRequired
};

ScoreControls.defaultProps = {
  isBeatOn: true,
  isTextOn: true,
  isPercussionOn: true,
  isNohkanOn: true,
  isDanceOn: true,
  isPrevSentenceOn: true,
  isNextSentenceOn: true,
  updateScoreToggles: () => {},
  startTime: 0
};

const mapDispatchToProps = dispatch => ({
  updateScoreToggles: toggles => dispatch(setScoreToggles(toggles))
});

export const Unwrapped = ScoreControls;
export default connect(
  null,
  mapDispatchToProps
)(ScoreControls);
