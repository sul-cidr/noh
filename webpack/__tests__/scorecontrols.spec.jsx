import React from "react";
import { Provider } from "react-redux";
import { shallow, mount } from "enzyme";
import configureMockStore from "redux-mock-store";
import ScoreControls, {
  Unwrapped as UnwrappedScoreControls
} from "../components/ScoreControls";

describe("<ScoreControls>", () => {
  let wrapper;
  let store;

  beforeEach(() => {
    const mockStore = configureMockStore();
    const initialState = {};
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <ScoreControls updateScoreToggles={jest.fn()} duration={100} />
      </Provider>
    );
  });

  it("renders as expected by default", () => {
    const component = shallow(<UnwrappedScoreControls duration={100} />);
    expect(component).toMatchSnapshot();
  });

  it("renders as expected by default with a store", () => {
    expect(wrapper).toMatchSnapshot();
  });

  fit("handles checkbox changes", () => {
    const action = { type: "SET_SCORE_TOGGLES", payload: {} };
    wrapper.find("input").forEach(input => {
      input.simulate("change");
      expect(store.getActions()[0].type).toEqual(action.type);
    });
  });
});
