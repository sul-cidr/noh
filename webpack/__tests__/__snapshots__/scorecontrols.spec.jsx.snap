// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScoreControls> renders as expected by default 1`] = `
<div
  className="score-controls"
>
  <div
    className="sentence-control"
  >
    <div
      className="sentence-control__prev"
    >
      <i
        className="fas fa-step-backward"
      />
    </div>
    <div
      className="sentence-control__status"
    >
      <span
        className="sentence-control__title"
      >
        Sentence:
      </span>
      <span
        className="sentence-control__current"
      >
        13/55
      </span>
    </div>
    <div
      className="sentence-control__next"
    >
      <i
        className="fas fa-step-forward"
      />
    </div>
  </div>
  <div
    className="video-progress"
  >
    <Connect(TimelineIndicator)
      duration={100}
      startTime={0}
    />
  </div>
  <div
    className="score-controls__filters"
  >
    <button
      className="score-controls__filters-button"
      onClick={[Function]}
    >
      Filters
    </button>
    <div
      className="score-controls__filters-popup hidden"
    >
      <ul
        className="channel-toggles"
      >
        <li
          className="toggles__title"
        >
          Lines
        </li>
        <li>
          <div
            className="custom-checkbox"
          >
            <input
              checked={true}
              id="scoreBeat"
              onChange={[Function]}
              onKeyPress={null}
              type="checkbox"
            />
            <label
              htmlFor="scoreBeat"
            >
              <span
                className="custom-checkbox__text"
              >
                Beat
              </span>
            </label>
          </div>
        </li>
        <li>
          <div
            className="custom-checkbox"
          >
            <input
              checked={true}
              id="scoreText"
              onChange={[Function]}
              onKeyPress={null}
              type="checkbox"
            />
            <label
              htmlFor="scoreText"
            >
              <span
                className="custom-checkbox__text"
              >
                Text
              </span>
            </label>
          </div>
        </li>
        <li>
          <div
            className="custom-checkbox"
          >
            <input
              checked={true}
              id="scorePercussion"
              onChange={[Function]}
              onKeyPress={null}
              type="checkbox"
            />
            <label
              htmlFor="scorePercussion"
            >
              <span
                className="custom-checkbox__text"
              >
                Percussion
              </span>
            </label>
          </div>
        </li>
        <li>
          <div
            className="custom-checkbox"
          >
            <input
              checked={true}
              id="scoreNohkan"
              onChange={[Function]}
              onKeyPress={null}
              type="checkbox"
            />
            <label
              htmlFor="scoreNohkan"
            >
              <span
                className="custom-checkbox__text"
              >
                Nohkan
              </span>
            </label>
          </div>
        </li>
        <li>
          <div
            className="custom-checkbox"
          >
            <input
              checked={true}
              id="scoreDance"
              onChange={[Function]}
              onKeyPress={null}
              type="checkbox"
            />
            <label
              htmlFor="scoreDance"
            >
              <span
                className="custom-checkbox__text"
              >
                Dance
              </span>
            </label>
          </div>
        </li>
      </ul>
      <ul
        className="measure-toggles"
      >
        <li
          className="toggles__title"
        >
          Sentences
        </li>
        <li>
          <div
            className="custom-checkbox"
          >
            <input
              checked={true}
              id="scorePrevSentence"
              onChange={[Function]}
              onKeyPress={null}
              type="checkbox"
            />
            <label
              htmlFor="scorePrevSentence"
            >
              <span
                className="custom-checkbox__text"
              >
                Previous Sentence
              </span>
            </label>
          </div>
        </li>
        <li>
          <div
            className="custom-checkbox"
          >
            <input
              checked={true}
              id="scoreNextSentence"
              onChange={[Function]}
              onKeyPress={null}
              type="checkbox"
            />
            <label
              htmlFor="scoreNextSentence"
            >
              <span
                className="custom-checkbox__text"
              >
                Next Sentence
              </span>
            </label>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`<ScoreControls> renders as expected by default with a store 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ScoreControls)
    duration={100}
    updateScoreToggles={[Function]}
  >
    <ScoreControls
      duration={100}
      isBeatOn={true}
      isDanceOn={true}
      isNextSentenceOn={true}
      isNohkanOn={true}
      isPercussionOn={true}
      isPrevSentenceOn={true}
      isTextOn={true}
      startTime={0}
      updateScoreToggles={[Function]}
    >
      <div
        className="score-controls"
      >
        <div
          className="sentence-control"
        >
          <div
            className="sentence-control__prev"
          >
            <i
              className="fas fa-step-backward"
            />
          </div>
          <div
            className="sentence-control__status"
          >
            <span
              className="sentence-control__title"
            >
              Sentence:
            </span>
            <span
              className="sentence-control__current"
            >
              13/55
            </span>
          </div>
          <div
            className="sentence-control__next"
          >
            <i
              className="fas fa-step-forward"
            />
          </div>
        </div>
        <div
          className="video-progress"
        >
          <Connect(TimelineIndicator)
            duration={100}
            startTime={0}
          >
            <TimelineIndicator
              currentTime={0}
              duration={100}
              interval={10}
              playing={false}
              startTime={0}
              updateCurrentTime={[Function]}
            >
              <div
                className="time-indicator-container"
              >
                <Draggable
                  allowAnyClick={false}
                  axis="x"
                  bounds="parent"
                  cancel={null}
                  defaultClassName="react-draggable"
                  defaultClassNameDragged="react-draggable-dragged"
                  defaultClassNameDragging="react-draggable-dragging"
                  defaultPosition={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  disabled={false}
                  enableUserSelectHack={true}
                  grid={null}
                  handle=".time-indicator"
                  offsetParent={null}
                  onDrag={[Function]}
                  onMouseDown={[Function]}
                  onStart={[Function]}
                  onStop={[Function]}
                  position={
                    Object {
                      "x": 0,
                      "y": 0,
                    }
                  }
                  transform={null}
                >
                  <DraggableCore
                    allowAnyClick={false}
                    axis="x"
                    bounds="parent"
                    cancel={null}
                    defaultClassName="react-draggable"
                    defaultClassNameDragged="react-draggable-dragged"
                    defaultClassNameDragging="react-draggable-dragging"
                    defaultPosition={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    disabled={false}
                    enableUserSelectHack={true}
                    grid={null}
                    handle=".time-indicator"
                    offsetParent={null}
                    onDrag={[Function]}
                    onMouseDown={[Function]}
                    onStart={[Function]}
                    onStop={[Function]}
                    position={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    transform={null}
                  >
                    <div
                      className="time-indicator react-draggable"
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "touchAction": "none",
                          "transform": "translate(0px,0px)",
                        }
                      }
                      transform={null}
                    />
                  </DraggableCore>
                </Draggable>
              </div>
            </TimelineIndicator>
          </Connect(TimelineIndicator)>
        </div>
        <div
          className="score-controls__filters"
        >
          <button
            className="score-controls__filters-button"
            onClick={[Function]}
          >
            Filters
          </button>
          <div
            className="score-controls__filters-popup hidden"
          >
            <ul
              className="channel-toggles"
            >
              <li
                className="toggles__title"
              >
                Lines
              </li>
              <li>
                <div
                  className="custom-checkbox"
                >
                  <input
                    checked={true}
                    id="scoreBeat"
                    onChange={[Function]}
                    onKeyPress={null}
                    type="checkbox"
                  />
                  <label
                    htmlFor="scoreBeat"
                  >
                    <span
                      className="custom-checkbox__text"
                    >
                      Beat
                    </span>
                  </label>
                </div>
              </li>
              <li>
                <div
                  className="custom-checkbox"
                >
                  <input
                    checked={true}
                    id="scoreText"
                    onChange={[Function]}
                    onKeyPress={null}
                    type="checkbox"
                  />
                  <label
                    htmlFor="scoreText"
                  >
                    <span
                      className="custom-checkbox__text"
                    >
                      Text
                    </span>
                  </label>
                </div>
              </li>
              <li>
                <div
                  className="custom-checkbox"
                >
                  <input
                    checked={true}
                    id="scorePercussion"
                    onChange={[Function]}
                    onKeyPress={null}
                    type="checkbox"
                  />
                  <label
                    htmlFor="scorePercussion"
                  >
                    <span
                      className="custom-checkbox__text"
                    >
                      Percussion
                    </span>
                  </label>
                </div>
              </li>
              <li>
                <div
                  className="custom-checkbox"
                >
                  <input
                    checked={true}
                    id="scoreNohkan"
                    onChange={[Function]}
                    onKeyPress={null}
                    type="checkbox"
                  />
                  <label
                    htmlFor="scoreNohkan"
                  >
                    <span
                      className="custom-checkbox__text"
                    >
                      Nohkan
                    </span>
                  </label>
                </div>
              </li>
              <li>
                <div
                  className="custom-checkbox"
                >
                  <input
                    checked={true}
                    id="scoreDance"
                    onChange={[Function]}
                    onKeyPress={null}
                    type="checkbox"
                  />
                  <label
                    htmlFor="scoreDance"
                  >
                    <span
                      className="custom-checkbox__text"
                    >
                      Dance
                    </span>
                  </label>
                </div>
              </li>
            </ul>
            <ul
              className="measure-toggles"
            >
              <li
                className="toggles__title"
              >
                Sentences
              </li>
              <li>
                <div
                  className="custom-checkbox"
                >
                  <input
                    checked={true}
                    id="scorePrevSentence"
                    onChange={[Function]}
                    onKeyPress={null}
                    type="checkbox"
                  />
                  <label
                    htmlFor="scorePrevSentence"
                  >
                    <span
                      className="custom-checkbox__text"
                    >
                      Previous Sentence
                    </span>
                  </label>
                </div>
              </li>
              <li>
                <div
                  className="custom-checkbox"
                >
                  <input
                    checked={true}
                    id="scoreNextSentence"
                    onChange={[Function]}
                    onKeyPress={null}
                    type="checkbox"
                  />
                  <label
                    htmlFor="scoreNextSentence"
                  >
                    <span
                      className="custom-checkbox__text"
                    >
                      Next Sentence
                    </span>
                  </label>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </ScoreControls>
  </Connect(ScoreControls)>
</Provider>
`;
